File: ts-proxy-tester-project/api/egress-ip.ts
import { BrowserContext, Page } from "playwright";

import { EgressProbe, probeEgressIp } from "../utils/egress";

interface Params {
  url: string;
}

export default async function handler(
  params: Params,
  page: Page,
  _context: BrowserContext
): Promise<EgressProbe> {
  return await probeEgressIp(page, params?.url);
}

File: py-proxy-tester-project/api/egress-ip.py
from playwright.async_api import BrowserContext, Page

from utils.egress import probe_egress_ip


async def automation(
    page: Page,
    params: dict | None = None,
    context: BrowserContext | None = None,
    **_kwargs,
) -> dict:
    if params is None or not params.get("url"):
        raise ValueError("This API requires a 'url' parameter")

    return dict(await probe_egress_ip(page, params["url"]))
